pytorch_lightning/trainer/trainer.py

```diff
@@ -1171,15 +1174,15 @@ def _run(
             self._call_callback_hooks("on_fit_end")
             self._call_lightning_module_hook("on_fit_end")
 
-        # teardown if necessary (similar calls for spawn plugins are excluded as they have
-        # been included at the end of `new_process` functions)
-        if not isinstance(self.training_type_plugin, DDPSpawnPlugin):
-            self._call_teardown_hook()
```

Comment:
Where was teardown hook get called for spawning before?

Reply:
directly in the DDPSpawnPlugin, here: https://github.com/PyTorchLightning/pytorch-lightning/blob/aeb0b5595fd73d086f4ae0f99d3f1f112f6a4c29/pytorch_lightning/plugins/training_type/ddp_spawn.py#L205

But now, this is no longer split.

Reply:
Oh I see, it teared down in the same function. Should it be after _collect_rank_zero_results then? same as comments below

Reply:
I prefer not to change the order in the Trainer._run() for existing behavior. 
It's not a big problem that teardown() is not the very very last hook to run. If we want to change the order, I believe we  should discuss the impact and do it in another PR.